test/functional/api/vim_spec.lua

```diff
@@ -81,6 +81,26 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_execute_lua', function()
```

Comment:
I don't know, it depends on interactions that I didn't evaluate. Also, adding test coverage exercises future accidental interactions.

> Also this is to some extent intended as an alternative to call_atomic in cases it is too primitive.

My intention is to recommend Lua for any server-side coordination logic, who knows what kind of things will be batched in call_atomic.